pkg/cmd/cli/cmd/newapp.go

```diff
@@ -511,6 +522,24 @@ func setLabels(labels map[string]string, result *newcmd.AppResult, ignoreFailure
 	return nil
 }
 
+func hasAppLabel(result *newcmd.AppResult) (bool, error) {
```

Comment:
Looks like all we do is set the label in a deployment config template:
https://github.com/openshift/origin/blob/master/pkg/util/labels.go#L175
so I guess we could just add that check here. However, it seems fragile. If later we add support for other internal objects in pkg/util/labels then we have to remember to fix this function.

The other thing is, why do we only process internal labels in DeploymentConfigs? Shouldn't we also handle anything else that can include a pod template?